Marlin/Marlin_main.cpp

```diff
@@ -1019,6 +1035,22 @@ inline void get_serial_commands() {
 
       // If command was e-stop process now
       if (strcmp(command, "M112") == 0) kill(PSTR(MSG_KILLED));
+      #if ENABLED(PROBE_FAIL_PANIC)
+        if (strcmp(command, "G26") == 0) {
+          LCD_MESSAGEPGM(WELCOME_MSG);
+          probe_fail = false;
+        }
+        else if(probe_fail && (strchr(command, 'G') != NULL || strchr(command, 'M') != NULL) && strstr(command, "M105") == NULL) {
+          for(uint8_t j=0; j < MAX_CMD_SIZE; j++)
+            command_queue[cmd_queue_index_r][j] = 0;
+          if(commands_in_queue) {
+            commands_in_queue = 1;
+            cmd_queue_index_r = (cmd_queue_index_r + 1) % BUFSIZE;
+          }
+          serial_count = 0;
+          return;
+        }
+      #endif
```

Comment:
We've had some discussions recently about the handling of commands ahead of the normal command buffer. I don't know how conclusive they were. But understand that if a host fills up the buffer after issuing `G29` (most likely while printing a GCode file) then the `G26` command may not be seen until the `G29` command exits and the machine resumes accepting commands.